setup.js

```diff
@@ -9,49 +14,29 @@ const mongod = new MongoMemoryServer(getMongodbMemoryOptions());
 const globalConfigPath = join(cwd, 'globalConfig.json');
 
 module.exports = async () => {
-  if (!mongod.isRunning) {
-    await mongod.start();
-  }
-
   const options = getMongodbMemoryOptions();
-  const mongoURLEnvName = getMongoURLEnvName();
 
-  const mongoConfig = {
-    mongoUri: await mongod.getUri(),
-    mongoDBName: options.instance.dbName,
-  };
+  const mongoConfig = {};
 
-  // Write global config to disk because all tests run in different contexts.
-  fs.writeFileSync(globalConfigPath, JSON.stringify(mongoConfig));
-  debug('Config is written');
+  //if we run one mongodb instance for all tests
+  if (!getSeparateMongoInstancesFlag()) {
+    if (!mongod.isRunning) {
+      await mongod.start();
+    }
 
-  // Set reference to mongod in order to close the server during teardown.
-  global.__MONGOD__ = mongod;
-  process.env[mongoURLEnvName] = mongoConfig.mongoUri;
-};
+    const mongoURLEnvName = getMongoURLEnvName();
 
-function getMongodbMemoryOptions() {
-  try {
-    const {mongodbMemoryServerOptions} = require(resolve(cwd, 'jest-mongodb-config.js'));
-
-    return mongodbMemoryServerOptions;
-  } catch (e) {
-    return {
-      binary: {
-        skipMD5: true,
-      },
-      autoStart: false,
-      instance: {},
-    };
-  }
-}
+    mongoConfig.mongoUri = await mongod.getUri();
 
-function getMongoURLEnvName() {
-  try {
-    const {mongoURLEnvName} = require(resolve(cwd, 'jest-mongodb-config.js'));
+    process.env[mongoURLEnvName] = mongoConfig.mongoUri;
 
-    return mongoURLEnvName || 'MONGO_URL';
-  } catch (e) {
-    return 'MONGO_URL';
+    // Set reference to mongod in order to close the server during teardown.
+    global.__MONGOD__ = mongod;
   }
-}
+
+  mongoConfig.mongoDBName = options.instance.dbName;
+
+  // Write global config to disk because all tests run in different contexts.
+  fs.writeFileSync(globalConfigPath, JSON.stringify(mongoConfig));
```

Comment:
Does config will be rewritten for every instance here?

Reply:
No, this is global setup that is triggered once before all test suites